content/en/platform/corda/4.12/enterprise/upgrade-guide.md

```diff
@@ -0,0 +1,265 @@
+
+Before upgrade:
+```
+.
+├── certificates
+├── corda-4.11.jar
+├── cordapps
+│   ├── config
+│   ├── corda-finance-contracts-4.11.jar
+│   └── corda-finance-workflows-4.11.jar
+├── drivers
+└── node.conf
+After upgrade
+```
+
+After upgrade:
+```
+.
+├── certificates
+├── corda-4.12.jar
+├── cordapps
+│   ├── config
+│   ├── corda-finance-contracts-4.12.jar
+│   └── corda-finance-workflows-4.12.jar
+├── drivers
+├── legacy-contracts
+│   └── corda-finance-contracts-4.11.jar
+└── node.conf
+```
+
+To ensure compatibility, you must keep the legacy contracts. When a node operating on a prior version of Corda 4.x wants to transact with a Corda 4.12 node, the 4.12 node identifies an old contract version attached to the transaction. To verify this old contract, Corda 4.12 initiates the External Verifier, which starts a new external process running Kotlin 1.2. This process facilitates transaction containing, for example, a Corda 4.11 contract to be verified on a Corda 4.12 node.
+
+Similarly, when a 4.12 node creates a transaction, it adds a 4.12 contract into a new component group of the transaction, reserving the existing component group for the 4.11 contract. Consequently, when a 4.11 contract gets attached to the transaction, it ends up with two sets of contract attachments (JARs): the legacy one and the new 4.12 contract. If this transaction is received by a node not running Corda 4.12, the node lacks awareness of the new component group. It disregards the 4.12 contract and proceeds to verify the legacy contract instead, which is then stored in the database.
+
+## Starting 4.12 nodes
+
+1. Run the database migration scripts.
+2. Start your node in the usual way:
+
+   ```
+   java -jar corda-4.12.jar -f node.conf
+   ```
+
+## Adding new 4.12 nodes
+
+If you want to add another Corda 4.12 node into your network post-upgrade, the specific steps vary depending on the state of your network.
+
+### Adding new 4.12 nodes to a mixed network
+
+If you are operating a mixed network, then the process for adding a new Corda 4.12 node is relatively straightforward.
+
+1. Set up the node folder structure same as other Corda 4.12 nodes.
+2. Add the `legacy-contracts` folder and associated files to the node folder.
+
+   ```
+   .
+   ├── certificates
+   ├── corda-4.12.jar
+   ├── cordapps
+   │   ├── config
+   │   ├── corda-finance-contracts-4.12.jar
+   │   └── corda-finance-workflows-4.12.jar
+   ├── drivers
+   ├── legacy-contracts
+   │   └── corda-finance-contracts-4.11.jar
+   └── node.conf
+   ```
+
+3. Register the node to the network and proceed to operate it as normal.
+
+### Adding new 4.12 nodes to a non-mixed network
+
+Perform the following steps if you are adding a new Corda 4.12 node to a network that is only running Corda 4.12 nodes, but also contains older Corda 4.x transactions.
+
+{{< note >}}
+In this scenario, you must keep a copy of the old CorDapp contract JAR file.
```

Comment:
The wording on this seems a bit off:

In this scenario, you still require a copy of the old CorDapp contract JAR file.

might sound better